public/components/datasources/components/manage/data_connection.tsx

```diff
@@ -294,6 +344,55 @@ export const DataConnection = (props: any) => {
     );
   };
 
+  useEffect(() => {
+    // Loads databases cache when cache is empty
+    if (datasourceDetails.name) {
+      const cache = CatalogCacheManager.getDataSourceCache().dataSources.find(
+        (datasource) => datasource.name === datasourceDetails.name
+      );
+      if (cache?.status !== 'Updated') {
+        startLoadingDatabases(datasourceDetails.name);
+        setIsRefreshing(true);
+      } else {
+        const cachedList: CachedDatabase[] =
+          CatalogCacheManager.getDataSourceCache().dataSources.find(
+            (cachedDataSource) => cachedDataSource.name === datasourceDetails.name
+          )?.databases || [];
```

Comment:
May be use `getDatabase` https://github.com/opensearch-project/dashboards-observability/blob/41214c22a67157b7a2f1d100162b99d4d6d5d498/public/framework/catalog_cache/cache_manager.ts#L135-L149

Reply:
I'm trying to get a list of databases here. I was under the impression that `getDatabase()` is used for retrieving a single database. I do use `getDatabase()` when attempting to retrieve a single database.